components/style/parser.rs

```diff
@@ -9,15 +9,32 @@ use selectors::parser::ParserContext as SelectorParserContext;
 use stylesheets::Origin;
 use url::Url;
 
+#[cfg(not(feature = "gecko"))]
+pub type ParserContextExtraData = ();
+
+#[cfg(feature = "gecko")]
+pub struct ParserContextExtraData {
+}
+
+#[cfg(not(feature = "gecko"))]
+pub fn dummy_extra_data() {}
+
+#[cfg(feature = "gecko")]
+pub fn dummy_extra_data() -> ParserContextExtraData {
+    ParserContextExtraData { }
+}
```

Comment:
I cant add a static method to ParserContextExtraData when it's a typedef for () in the non-geckolib config.  I could make it an enum with no constructors.  (And all the non-geckolib places where I pass in () for the extra_data parameter I'll replace with ParserContextExtraData::default()).